src/content/docs/en/extend/add-view-transitions.mdx

```diff
@@ -0,0 +1,351 @@
+
+    ```astro title="src/layouts/BaseLayout.astro" ins={2,15}
+    ---
+    import { ViewTransitions } from 'astro:transitions';
+    import Header from "../components/Header.astro";
+    import Footer from "../components/Footer.astro";
+    import "../styles/global.css";
+    const { pageTitle } = Astro.props;
+    ---
+    <html lang="en">
+      <head>
+        <meta charset="utf-8" />
+        <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+        <meta name="viewport" content="width=device-width" />
+        <meta name="generator" content={Astro.generator} />
+        <title>{pageTitle}</title>
+        <ViewTransitions />
+      </head>
+      <body>
+        <Header />
+        <h1>{pageTitle}</h1>
+        <slot />
+        <Footer />
+        <script>
+          import "../scripts/menu.js";
+        </script>
+      </body>
+    </html>
+    ```
+
+    No other configuration is necessary to enable Astro's default client-side navigation! Astro will create default page animations based on the similarities between the old and new page, and will also provide fallback behavior for unsupported browsers.
+
+3. Navigate between pages in your site preview
+
+    As you move between pages on your site, notice that the old page content appears to fade out as the new page content fades in. You can now add custom behavior to control individual page links if you are not satisfied with the defaults.
+
+### Update scripts
+
+With view transitions, some scripts may no longer re-run after page navigation like they do with full-page browser refreshes. There are two events during client-side routing that you can listen for, and fire events when they occur: [`astro:page-load`](/en/guides/view-transitions/#astropage-load) and [`astro:after-swap`](/en/guides/view-transitions/#astroafter-swap).
+
+4. Make the script controling the `<Hamburger />` mobile menu component available after navigating to a new page.
+
+     To make your mobile menu interactive after navigating to a new page, add the following code that listens for the `astro:page-load` event which runs at the end of page navigation, and in response, runs the existing script to make the hamburger menu function when clicked:
+
+    ```js title="src/scripts/menu.js" ins={1,5}
+    document.addEventListener('astro:page-load', () => {
+      document.querySelector('.hamburger').addEventListener('click', () => {
+        document.querySelector('.nav-links').classList.toggle('expanded');
+      });
+    });
+    ```
+
+5. Make the script controlling the theme toggle available after page navigation.
+
+    The `<script>` that controls the light/dark theme toggle is located in the `<ThemeIcon />` component. For the theme toggle to continue to function on every page, remove the `is:inline` attribute from the script add the same event listener as in the previous example so that `astro:page-load` event can trigger your existing function.
```

Comment:
```suggestion
    The `<script>` that controls the light/dark theme toggle is located in the `<ThemeIcon />` component. For the theme toggle to continue to function on every page, remove the `is:inline` attribute from the script and add the same event listener as in the previous example so that `astro:page-load` event can trigger your existing function.
```

I think this may be missing an "and" or a comma between the 2 actions.